tests/serverpod_test_server/test/client_models_test.dart

```diff
@@ -0,0 +1,234 @@
+import 'dart:typed_data';
+
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:test/test.dart';
+
+ByteData createByteData(int len) {
+  var data = Uint8List(len);
+  for (var i = 0; i < len; i++) {
+    data[i] = i % 256;
+  }
+  return ByteData.view(data.buffer);
+}
+
+void main() {
+  setUp(() {});
+
+  group('client model "operator=="', () {
+    test('ByteData not equal', () {
+      var a = createByteData(2);
+      var b = createByteData(2);
+
+      expect(a, isNot(b));
+    });
+
+    group('SimpleData', () {
+      test('equal', () {
+        expect(SimpleData(num: 0), equals(SimpleData(num: 0)));
+        expect(SimpleData(num: 1, id: 1), equals(SimpleData(num: 1, id: 1)));
+      });
+
+      test('not equal', () {
+        expect(SimpleData(num: 0), isNot(SimpleData(num: 10)));
+        expect(SimpleData(num: 1, id: 1), isNot(SimpleData(num: 2, id: 2)));
+      });
+    });
+
+    group('SimpleDataList', () {
+      var a = SimpleDataList(rows: [
+        SimpleData(id: 1, num: 1),
+      ]);
+
+      test('equal', () {
+        var b = a.copyWith(rows: [
+          SimpleData(id: 1, num: 1),
+        ]);
+
+        expect(a, equals(b));
+      });
+
+      test('not equal', () {
+        var b = a.copyWith(rows: [
+          SimpleData(id: 2, num: 2),
+        ]);
+        expect(a, isNot(b));
+      });
+    });
+
+    group('ObjectWithObject', () {
```

Comment:
Like the separation! But I still think the test descriptions are quite vague. If you run these tests without the context of this PR it will be hard to understand by just reading the test description why something failed. 

I.E. make it clear what we are testing in each test, if you look at the structure in the original comment. 

We first make a group of what we are testing.
Then describing what each test is checking.

Does it make sense?

Would be nice with a test with an empty copyWith(); too!